kernel/build/config.rs

```diff
@@ -19,7 +19,77 @@
 //! This file implements the configuration file for compilation.
 
 use serde::Deserialize;
-use std::{fs, io};
+use std::{any::type_name, fs, io, process::exit};
+
+/// Generate a cfg flag if the value is true.
+macro_rules! generate_cfg_flag {
+	($value:expr) => {
+		if $value {
+			let full_path = stringify!($value);
+			let cfg_name = format!(
+				"config_{}",
+				full_path
+					.trim_start_matches("&")
+					.split('.')
+					.skip_while(|s| *s == "self")
+					.collect::<Vec<_>>()
+					.join("_")
+			);
+
+			println!("cargo:rustc-cfg={}", cfg_name);
+		}
+	};
+}
+
+/// Generate a Rust const file in OUT_DIR.
+macro_rules! generate_const_file {
+	($value:expr) => {{
+		fn inner<T: std::fmt::Debug + std::fmt::Display>(value: T, name: &str) {
+			let type_suffix = match type_name::<T>() {
+				"u64" => "u64",
+				"u32" => "u32",
+				"i64" => "i64",
+				"i32" => "i32",
+				"bool" => "",
+				"alloc::string::String" | "&str" | "&alloc::string::String" => "",
+				t => {
+					eprintln!("Unsupported config type {t}");
+					exit(1);
+				}
+			};
```

Comment:
I think the type suffix isn't necessary? I am pretty sure Rust can infer it itself